src/nebula-hooks/virtualized-table/use-wait-for-fonts.ts

```diff
@@ -0,0 +1,29 @@
+import { usePromise } from '@nebula.js/stardust';
+import { getBodyStyle, getHeaderStyle } from '../../table/utils/styling-utils';
+import { ExtendedTheme, TableLayout } from '../../types';
+
+const useWaitForFonts = (theme: ExtendedTheme, layout: TableLayout, shouldRender: boolean) => {
```

Comment:
If font family is changed between 2 renders. For example by changing theme. This will first return `true` before it has checked that the new font has been loaded.

It a scenario that I think can happen, if a font is somehow dynamically loaded between the two renders. But I do find it unlikely to happen so I did end up not added some additional logic to handle this case.